$OpenBSD: patch-src_video_out_video_out_opengl_c,v 1.1 2012/04/07 05:48:49 ajacoutot Exp $

Quick-fix to enable OSD when using fragment shader for YUV conversion.

--- src/video_out/video_out_opengl.c.orig	Fri Feb 17 23:24:07 2012
+++ src/video_out/video_out_opengl.c	Fri Feb 17 23:25:19 2012
@@ -1480,6 +1480,20 @@ static void opengl_overlay_blend (vo_driver_t *this_ge
         XUnlockDisplay (this->display);
       }
     } else {
+
+      if (!frame->rgb_dst) {
+        if (frame->format == XINE_IMGFMT_YV12) {
+          _x_blend_yuv(frame->vo_frame.base, overlay,
+                       frame->width, frame->height, frame->vo_frame.pitches,
+                       &this->alphablend_extra_data);
+        } else {
+          _x_blend_yuy2(frame->vo_frame.base[0], overlay,
+                        frame->width, frame->height, frame->vo_frame.pitches[0],
+                        &this->alphablend_extra_data);
+        }
+        return;
+      }
+
       if (!overlay->rgb_clut || !overlay->hili_rgb_clut)
         opengl_overlay_clut_yuv2rgb (this, overlay, frame);
 
